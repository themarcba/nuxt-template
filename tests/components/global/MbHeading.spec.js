import {} from 'jest';
// Doc: https://vue-test-utils.vuejs.org/api/
import { createLocalVue, config, shallowMount } from '@vue/test-utils';
import MbHeading from '@/components/global/MbHeading.vue';
import NuxtStubs from '@/tests/nuxt-stubs';

const localVue = createLocalVue();

// Doc: https://vue-test-utils.vuejs.org/api/#config
config.showDeprecationWarnings = true;

describe('MbHeading.vue', () => {
  const wrapper = shallowMount(MbHeading, {
    localVue,
    stubs: NuxtStubs
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it('matches the snapshot', async () => {
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ level: 1 });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ level: 2, accent: true, primary: false, secondary: false });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ level: 3, accent: false, primary: true, secondary: false });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ level: 4, accent: false, primary: false, secondary: true });
    expect(wrapper).toMatchSnapshot();
  });
});
