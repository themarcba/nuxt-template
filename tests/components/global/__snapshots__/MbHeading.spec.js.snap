// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MbHeading.vue matches the snapshot 1`] = `<h1 class="font-bold text-2xl"></h1>`;

exports[`MbHeading.vue matches the snapshot 2`] = `<h1 class="font-bold text-2xl"></h1>`;

exports[`MbHeading.vue matches the snapshot 3`] = `<h2 class="font-bold text-accent text-xl"></h2>`;

exports[`MbHeading.vue matches the snapshot 4`] = `<h3 class="font-bold text-primary text-lg"></h3>`;

exports[`MbHeading.vue matches the snapshot 5`] = `<h4 class="font-bold text-secondary text-base"></h4>`;
